rsocket-core/src/main/java/io/rsocket/core/RSocketRequester.java

```diff
@@ -306,10 +313,67 @@ public void accept(long n) {
       return Flux.error(err);
     }
 
+    return request.switchOnFirst(
+        (s, flux) -> {
+          Payload payload = s.get();
+          if (payload != null) {
+            return handleChannel(payload, flux);
+          } else {
+            return flux;
+          }
+        },
+        false);
+  }
+
+  private Flux<? extends Payload> handleChannel(Payload initialPayload, Flux<Payload> inboundFlux) {
     final UnboundedProcessor<ByteBuf> sendProcessor = this.sendProcessor;
-    final UnicastProcessor<Payload> receiver = UnicastProcessor.create();
+    final AtomicBoolean payloadReleasedFlag = new AtomicBoolean(false);
     final int streamId = streamIdSupplier.nextStreamId(receivers);
 
+    final UnicastProcessor<Payload> receiver = UnicastProcessor.create();
+    final BaseSubscriber<Payload> upstreamSubscriber =
+        new BaseSubscriber<Payload>() {
+
+          boolean first = true;
+
+          @Override
+          protected void hookOnSubscribe(Subscription subscription) {
+            // noops
+          }
+
+          @Override
+          protected void hookOnNext(Payload payload) {
+            if (first) {
+              // need to skip first since we have already sent it
+              first = false;
+              return;
+            }
```

Comment:
The `if (first)` check could be replaced with `inboundFlux.skip(1)`?

Reply:
It was with skip initially, but, I figured out that skip does extra request(1) when it skips, so we have extra 1 element requested (I mentioned that in the comments that expected prefetch is 256, but it was 257)

The strategy is the following 

1) we do at the very beginning `request(1)` (from the `switchOnFirst`) to obtain the first element and then do the transformation. 
2) Then, we send that obtained element as a part of `RequestChannel` frame so it appears on the moment of delivery
3) Then, in order to no brake backpressure because of the eager first element prefetch, we reduce 1 from the first upcoming request to the `switchOnFirst`.
4) However, the first prefetched element is sent to the flux along the way, and that would be OK if we have just normal routing case, but, in case of RSocket, the first payload has already been sent so if we are not going to `skip` the first element, it will be delivered twice
5) Here comes the issue, if we use `skip` then `skip` operator think that we need to fulfill the demand by requesting extra one element, but in fact we just need not skip but `DROP` element instead without doing `request(1)`